modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/service/FriendlyURLDLFileEntryLocalServiceWrapper.java

```diff
@@ -0,0 +1,188 @@
+import com.liferay.dynamic.data.mapping.kernel.DDMFormValues;
+import com.liferay.friendly.url.model.FriendlyURLEntry;
+import com.liferay.friendly.url.service.FriendlyURLEntryLocalService;
+import com.liferay.portal.configuration.metatype.bnd.util.ConfigurableUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.repository.model.FileEntry;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceContextThreadLocal;
+import com.liferay.portal.kernel.service.ServiceWrapper;
+import com.liferay.portal.kernel.util.LocaleUtil;
+
+import java.io.File;
+import java.io.InputStream;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alicia García
+ */
+@Component(
+	configurationPid = "com.liferay.document.library.configuration.FFFriendlyURLEntryFileEntryConfiguration",
+	service = ServiceWrapper.class
+)
+public class FriendlyURLDLFileEntryLocalServiceWrapper
+	extends DLFileEntryLocalServiceWrapper {
+
+	@Override
+	public DLFileEntry addFileEntry(
+			String externalReferenceCode, long userId, long groupId,
+			long repositoryId, long folderId, String sourceFileName,
+			String mimeType, String title, String description, String changeLog,
+			long fileEntryTypeId, Map<String, DDMFormValues> ddmFormValuesMap,
+			File file, InputStream inputStream, long size, Date expirationDate,
+			Date reviewDate, ServiceContext serviceContext)
+		throws PortalException {
+
+		DLFileEntry dlFileEntry = super.addFileEntry(
+			externalReferenceCode, userId, groupId, repositoryId, folderId,
+			sourceFileName, mimeType, title, description, changeLog,
+			fileEntryTypeId, ddmFormValuesMap, file, inputStream, size,
+			expirationDate, reviewDate, serviceContext);
+
+		if (_ffFriendlyURLEntryFileEntryConfiguration.enabled()) {
+			_addFriendlyURLEntry(dlFileEntry, title);
+		}
+
+		return dlFileEntry;
+	}
+
+	@Override
+	public DLFileEntry deleteFileEntry(long fileEntryId)
+		throws PortalException {
+
+		DLFileEntry dlFileEntry = super.deleteFileEntry(fileEntryId);
+
+		if (_ffFriendlyURLEntryFileEntryConfiguration.enabled()) {
+			List<FriendlyURLEntry> friendlyURLEntries =
+				_friendlyURLEntryLocalService.getFriendlyURLEntries(
+					dlFileEntry.getGroupId(),
+					_classNameLocalService.getClassNameId(FileEntry.class),
+					dlFileEntry.getFileEntryId());
+
+			for (FriendlyURLEntry friendlyURLEntry : friendlyURLEntries) {
+				_friendlyURLEntryLocalService.deleteFriendlyURLEntry(
+					friendlyURLEntry);
+			}
+		}
```

Comment:
There's a `deleteFriendlyURLEntry(groupId, className, classPK)` method that does this; there's no need to iterate explicitly (sorry I didn't see this in the previous review).

Reply:
👍🏻